shopify_function_macro/src/lib.rs

```diff
@@ -127,14 +128,73 @@ pub fn generate_types(attr: proc_macro::TokenStream) -> proc_macro::TokenStream
         #[graphql(
             query_path = #OUTPUT_QUERY_FILE_NAME,
             schema_path = #schema_path,
-            response_derives = "Clone,Debug,PartialEq,Eq,Deserialize",
-            variables_derives = "Clone,Debug,PartialEq,Eq,Deserialize",
+            response_derives = "Clone,Debug,PartialEq,Eq,Deserialize,OneOfDefault",
+            variables_derives = "Clone,Debug,PartialEq,Eq,Deserialize,OneOfDefault",
             skip_serializing_none
         )]
         struct Output;
     }
     .into()
 }
 
+fn is_type(ty: &syn::Type, name: impl AsRef<str>) -> bool {
+    if let syn::Type::Path(path) = ty {
+        path.path.segments.len() == 1
+            && path
+                .path
+                .segments
+                .first()
+                .unwrap()
+                .ident
+                .to_string()
+                .as_str()
+                == name.as_ref()
+    } else {
+        false
+    }
+}
+
+#[proc_macro_derive(OneOfDefault)]
+pub fn one_of_default_derive(item: proc_macro::TokenStream) -> proc_macro::TokenStream {
+    let parsed_item: syn::Item = parse_macro_input!(item);
+    let syn::Item::Struct(parsed_struct) = parsed_item else { return quote!{}.into()};
+    // If every field in the struct is an `Option`, it's the result
+    // of a `@oneOf` directive.
+    let is_oneof_struct = parsed_struct
+        .fields
+        .iter()
+        .all(|field| is_type(&field.ty, "Option"));
```

Comment:
Not necessarily true, I believe? (sorry if I'm being pedantic) But nonetheless it does mean that we can derive `Default`